include/Actuator.h

```diff
@@ -66,6 +99,105 @@ class Actuator
   // populate nodal field and output norms (if appropriate)
   virtual void execute() = 0;
 
+  // Use for polymorphic print statements
+  virtual std::string get_class_name() = 0;
+
+//------------------------------------------------------------------
+
+  // determine element bounding box in the mesh
+  void populate_candidate_elements();
+
+  // fill in the map that will hold point and ghosted elements
+  void create_actuator_line_point_info_map();
+
+  // figure out the set of elements that belong in the custom ghosting data structure
+  void determine_elems_to_ghost();
+
+  // deal with custom ghosting
+  void manage_ghosting();
+
+  // populate vector of elements
+  void complete_search();
+
+  // support methods to gather data; scalar and vector
+  void resize_std_vector(
+    const int &sizeOfField,
+    std::vector<double> &theVector,
+    stk::mesh::Entity elem,
+    const stk::mesh::BulkData & bulkData);
+
+  // general gather methods for scalar and vector (both double)
+  void gather_field(
+    const int &sizeOfField,
+    double *fieldToFill,
+    const stk::mesh::FieldBase &stkField,
+    stk::mesh::Entity const* elem_node_rels,
+    const int &nodesPerElement);
+
+  void gather_field_for_interp(
+    const int &sizeOfField,
+    double *fieldToFill,
+    const stk::mesh::FieldBase &stkField,
+    stk::mesh::Entity const* elem_node_rels,
+    const int &nodesPerElement);
+
+  // element volume and scv volume populated
+  double compute_volume(
+    const int &nDim,
+    stk::mesh::Entity elem,
+    const stk::mesh::BulkData & bulkData);
+
+  // interpolate field to point centroid
+  void interpolate_field(
+    const int &sizeOfField,
+    stk::mesh::Entity elem,
+    const stk::mesh::BulkData & bulkData,
+    const double *isoParCoords,
+    const double *fieldAtNodes,
+    double *pointField);
+
+  double compute_distance(
+    const int &nDim,
+    const double *elemCentroid,
+    const double *pointCentroid);
+/*
+  // distance from element centroid to point centroid
+  double compute_distance(
+    const int &nDim,
+    const double *elemCentroid,
+    const double *pointCentroid);
+*/
+//------------------------------------------------------------------
+  // hold the realm
+  Realm &realm_;
+
+  // type of stk search
+  stk::search::SearchMethod searchMethod_;
+
+  // how many elements to ghost?
+  uint64_t needToGhostCount_;
+  stk::mesh::EntityProcVec elemsToGhost_;
+
+  // save off product of search
+  std::vector<std::pair<theKey, theKey> > searchKeyPair_;
+  // bounding box data types for stk_search */
+  std::vector<boundingSphere> boundingSphereVec_;
+  std::vector<boundingElementBox> boundingElementBoxVec_;
+  // target names for set of bounding boxes
+  std::vector<std::string> searchTargetNames_;
+
+  std::vector<ActuatorInfo*> actuatorInfo_;
```

Comment:
Would there be an issue using `std::unique_ptr` here? That will eliminate the code in destructor.

Reply:
Yes definitely.  I noticed this yesterday when I started working on the actuator disk.  I will make the change and push an update.